### PR TITLE
Implement replies

### PR DESCRIPTION
**Pull request by samoht**
And that seems to mostly work; known issues:

- Sometimes the first connection is not flushed properly and is blocking
- Accessing the underlying block storage seems to be broken currently (we get `Got an error: device is disconnected.` errors)

### CODE REVIEW

File: proto/lambda.proto [hunk 1]
```diff
@@ -161,6 +168,8 @@ message Value {
         Bytes bytes = 11;
         Result result = 12;
         Either either = 13;
+        Return return = 14;
+        Abstract abstract = 15;
```
Comment (aherrmann) [inline proto/lambda.proto]: What's the reasoning for allowing values of abstract type?

File: proto/lambda_protobuf.ml [hunk 2]
```diff
@@ -262,6 +264,8 @@ let rec value_from : Types.value -> value = function
      | None -> Fmt.invalid_arg "Cannot unify %a and %a"
                  Lambda.Type.pp t
                  Lambda.Type.pp terror)
+  | Types.Return { value; _ } -> value_from value
+  | Types.Abstract -> invalid_arg "Impossible to un-serialize an abstract value"
```
Comment (aherrmann) [inline proto/lambda_protobuf.ml]: If this is an error, wouldn't it be better to omit values of abstract type altogether?

File: src/typedtree.ml [hunk 3]
```diff
@@ -574,6 +579,7 @@ module Expr = struct
     | Prm x            -> P.prim (Prim.untype x)
     | Uno (Fst, x)     -> P.fst (untype x)
     | Uno (Snd, x)     -> P.snd (untype x)
+    | Uno (Get i, x)   -> P.get i (untype x)
```
Comment (aherrmann) [inline src/typedtree.ml]: Making `Get` a unary operator that holds the index means that the index must be statically known. So, it's impossible to write something like this:
```
\ (i : Int) -> \ (xs : List Bool) -> Get i xs
```
(Unless I'm missing something)
What's the reasoning behind that?

If `Get` was a binary operator taking the index and the list as arguments, then dynamic indices would be possible.

File: proto/lambda_protobuf.ml (hunk 2 above)
Comment (dinosaure) [reply proto/lambda_protobuf.ml]: I think it's better to delete it too. However, the server will respond an error if it wants to serialize an abstract type. In this case, we can serialize an abstract type and the client can deal with it differently.

File: src/typedtree.ml (hunk 3 above)
Comment (samoht) [reply src/typedtree.ml]: That was a quick hack to test the demo app, I'll push something better. Thanks for noticing :-)

**Comment by samoht** [comment]
@aherrmann I've just added a `prj` operator to project result values into either values. Let me know if that's enough for you.

**Comment by aherrmann** [comment]
> I've just added a prj operator to project result values into either values. Let me know if that's enough for you.

Great, thanks! That should enable programs to determine if something went wrong. With additional primitives for the error types (see #22) they could also return a meaningful error message.

**Pull request merged**